File: server/src/main/kotlin/it/polito/server/security/saml/JwtAuthConverterProperties.java
package it.polito.server.security.saml;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.validation.annotation.Validated;

@Validated
@Configuration
@ConfigurationProperties(prefix = "jwt.auth.converter")
@Getter
@Setter
public class JwtAuthConverterProperties {
    private String resourceId;
    private String principalAttribute;

}

File: server/src/main/kotlin/it/polito/server/security/JwtAuthConverterProperties.kt
package it.polito.server.security

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

import org.springframework.validation.annotation.Validated


@Validated
@Configuration
@ConfigurationProperties(prefix = "jwt.auth.converter")
class JwtAuthConverterProperties {
    var resourceId: String? = null
    var principalAttribute: String? = null
}
